pandas/core/indexes/range.py

```diff
@@ -118,6 +118,8 @@ def __new__(
         if isinstance(start, RangeIndex):
             start = start._range
             return cls._simple_new(start, name=name)
```

Comment:
If the above two lines were replaced with `return start.copy(name=name)`, the `_cache`attribute would be reused on the new object and we'd get a performance improvement in cases where we'd have to re-access the cache.

Reply:
neat, will update